test/fuzz/utility.h

```diff
@@ -76,5 +76,45 @@ inline TestStreamInfo fromStreamInfo(const test::fuzz::StreamInfo& stream_info)
   return test_stream_info;
 }
 
+// The HeaderMap code assumes that input does not contain certain characters, and
+// this is validated by the HTTP parser. Some fuzzers will create strings with
+// these characters, however, and this creates not very interesting fuzz test
+// failures as an assertion is rapidly hit in the LowerCaseString constructor
+// before we get to anything interesting.
+//
+// This method will replace any of those characters found with spaces.
+inline std::string replaceInvalidCharacters(absl::string_view string) {
+  std::string filtered;
+  filtered.reserve(string.length());
+  for (const char& c : string) {
+    switch (c) {
+    case '\0':
+      FALLTHRU;
+    case '\r':
+      FALLTHRU;
+    case '\n':
+      filtered.push_back(' ');
+      break;
+    default:
+      filtered.push_back(c);
+    }
+  }
+  return filtered;
+}
+
+// Return a new RepeatedPtrField of HeaderValueOptions with invalid characters removed.
+inline Protobuf::RepeatedPtrField<envoy::api::v2::core::HeaderValueOption> processHeadersToAdd(
```

Comment:
Nit: prefer something more descriptive like `replaceInvalidHeaders` (or whatever you prefer)